store-rx2/src/main/kotlin/com/dropbox/store/rx2/RxStore.kt

```diff
@@ -0,0 +1,96 @@
+package com.dropbox.store.rx2
+
+import com.dropbox.android.external.store4.BuilderImpl
+import com.dropbox.android.external.store4.BuilderWithSourceOfTruth
+import com.dropbox.android.external.store4.ExperimentalStoreApi
+import com.dropbox.android.external.store4.Store
+import com.dropbox.android.external.store4.StoreBuilder
+import com.dropbox.android.external.store4.StoreRequest
+import com.dropbox.android.external.store4.StoreResponse
+import io.reactivex.Flowable
+import io.reactivex.Maybe
+import io.reactivex.Single
+import kotlinx.coroutines.reactive.asFlow
+import kotlinx.coroutines.rx2.asFlowable
+import kotlinx.coroutines.rx2.await
+import kotlinx.coroutines.rx2.rxSingle
+
+/**
+ * Return a flow for the given key
+ * @param request - see [StoreRequest] for configurations
+ */
+fun <Key : Any, Output : Any> Store<Key, Output>.observe(request: StoreRequest<Key>): Flowable<StoreResponse<Output>> =
+    stream(request).asFlowable()
+
+/**
+ * Purge a particular entry from memory and disk cache.
+ * Persistent storage will only be cleared if a delete function was passed to
+ * [StoreBuilder.persister] or [StoreBuilder.nonFlowingPersister] when creating the [Store].
+ */
+fun <Key : Any, Output : Any> Store<Key, Output>.observeClear(key: Key): Single<Unit> =
```

Comment:
gosh these are ugly but i understand we cannot use the same names. 

should we consider completely wrapping this so that there is another class called RxStore so that we can use the more normal names there?

Reply:
lol that's what mike stared with. I suggested not adding another type to keep the API smaller

Reply:
```suggestion
fun <Key : Any, Output : Any> Store<Key, Output>.observeClear(key: Key): Completable =
```

Reply:
Ha that was the original example. @eyalgu and you can battle it out for extension vs wrapper class. I'm fine with both :-)

Reply:
I prefer extension over new classes. If you don't introduce a new type a future migration is easier (and you can also migrate gradually)

That being said, we might need a wrapper store in order to handle scoping correctly though maybe we can still avoid it somehow 🤷‍♂